x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentToken.java

```diff
@@ -0,0 +1,208 @@
+    protected CreateEnrollmentToken(Environment environment, CommandLineHttpClient client) {
+
+        this.environment = environment;
+        this.sslService = new SSLService(environment);
+        this.client = client;
+    }
+
+    public String create(Terminal terminal, String user, SecureString password) throws Exception {
+        final KeyConfig keyConfig = sslService.getHttpTransportSSLConfiguration().keyConfig();
+        if (keyConfig instanceof StoreKeyConfig == false) {
+            throw new UserException(ExitCodes.CONFIG,
+                "Unable to create an enrollment token. Elasticsearch node HTTP layer SSL configuration is not configured with a keystore");
+        }
+        final List<Tuple<PrivateKey, X509Certificate>> httpCaKeysAndCertificates =
+            ((StoreKeyConfig) keyConfig).getPrivateKeyEntries(environment).stream()
+                .filter(t -> t.v2().getBasicConstraints() != -1).collect(Collectors.toList());
+        if (httpCaKeysAndCertificates.isEmpty()) {
+            throw new UserException(ExitCodes.CONFIG,
+                "Unable to create an enrollment token. Elasticsearch node HTTP layer SSL configuration Keystore doesn't contain any " +
+                    "PrivateKey entries where the associated certificate is a CA certificate");
+        } else if (httpCaKeysAndCertificates.size() > 1) {
+            throw new UserException(ExitCodes.CONFIG,
+                "Unable to create an enrollment token. Elasticsearch node HTTP layer SSL configuration Keystore contain multiple " +
+                    "PrivateKey entries where the associated certificate is a CA certificate");
+        } else {
+            final CheckedSupplier<String, Exception> createApiKeyRequestBodySupplier = () -> {
+                XContentBuilder xContentBuilder = JsonXContent.contentBuilder();
+                xContentBuilder.startObject()
+                    .field("name", "enrollment_token_API_key_" + UUIDs.base64UUID())
+                    .field("expiration", ENROLL_API_KEY_EXPIRATION_SEC)
+                    .startObject("role_descriptors")
+                    .startObject("create_enrollment_token")
+                    .field("cluster", "[" + ClusterPrivilegeResolver.ENROLL_NODE.name() + "]")
+                    .endObject()
+                    .endObject()
+                    .endObject();
+                return Strings.toString(xContentBuilder);
+            };
+
+            final CheckedSupplier<String, Exception> getHttpInfoRequestBodySupplier = () -> {
+                XContentBuilder xContentBuilder = JsonXContent.contentBuilder();
+                xContentBuilder.startObject()
+                    .endObject();
+                return Strings.toString(xContentBuilder);
+            };
+
+            try {
+                final URL url = new URL(client.getDefaultURL());
+                final HttpResponse httpResponseApiKey = client.execute("POST", createAPIKeyURL(url), user, password,
```

Comment:
nit suggestion use a local var for the url instead of calling createAPIKeyURL twice ?